### ISSUE TITLE
Create Pipeline

### ISSUE THREAD

**Issue by kpfaulkner** [microsoft/azure-devops-go-api#79]
Hi

I'm trying to create a pipeline (well strictly speaking, clone a pipeline for backup purposes).
I can 'get' the a pipeline easily enough, has all my build steps, variables etc etc..  and I want to try and recreate the pipeline elsewhere. All of this is for backup/restore purposes.

My problem is the pipelines.CreatePipeline() function.  The CreatePipelineArgs doesn't seem to have much in it. There is an InputParameter parameter, which in turn has a CreatePipelineConfigurationParameters instance (which again doesn't seem to have much there).  I was picturing that somewhere in the payload for the CreatePipeline I would either be populating structures that are basically identical to what I get on the GetPipeline(), or accept JSON.  But I'm wrong.

Can someone clarify if creation of pipeline is working?

Thanks

Ken

**Comment by raboley** [comment]
Was wondering if writing my pipelines in go was possible and based on this issue that seems to be most likely not true. Is that correct?

**Comment by kpfaulkner** [comment]
@raboley Currently unsure. No success myself... was hoping someone at MS might be able to answer the question here.

**Comment by raboley** [comment]
> @raboley Currently unsure. No success myself... was hoping someone at MS might be able to answer the question here.

I switched to just writing a go cli tool that holds all business logic, and then my pipeline just installs and runs commands from that tool to minimize the logic embedded in the pipelines.

**Comment by summitdevops** [comment]
pipeClient.CreatePipeline is not working , i am getting following error and i dont see any Path parameter.
 Value cannot be null.
Parameter name: Path
exit status 1

@kpfaulkner  did you able to fix the issue?

**Comment by yradsmikham** [comment]
Agreed with all the above comments. `CreatePipeline` does not appear to be in a mature state as it is missing: `path`, `repository.id`, `repository.name`, and `repository.type`.

Here's a blog that covers it more in detail: https://blog.johnnyreilly.com/2021/05/08/create-pipeline-with-azure-devops-api 

Any update on this?

**Comment by AdamRussak** [comment]
same here, is it marked as an bug/error? or are we missing something?
